File: LeetCode/LeetCode_Easy/C#/TwoSum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) 
    {
        var dict = new Dictionary<int, int>();

        for (int i = 0; i < nums.Length; i++)
        {
            int diff = target - nums[i];
            
            if (dict.TryGetValue(nums[i], out var dictValue)) return new int[] { dictValue, i };

            dict.TryAdd(diff, i);
        }

        return new int[] {};
    } 
}  
    

File: LeetCode/LeetCode_Easy/C++/TwoSum.cpp
class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {
        std::unordered_map<int, int> map;

        for (int i = 0; i < nums.size(); i++) {
            int keyOfMap = target - nums[i];
            auto iter = map.find(keyOfMap);

            if (iter != map.end()) {
                return {iter->second, i};
            }

            map[nums[i]] = i;
        }
        return {};
    }
};
